backend/server/tests/planner/test_validate_term_planner.py

```diff
@@ -180,16 +180,14 @@ def test_validateTermPlanner_past_term_suppress_warnings():
             "is_accurate": True,
             "unlocked": False,
             "handbook_note": "",
-            "warnings": [],
+            "warnings": ['((COMP1531 AND (COMP2521 OR COMP1927)))'],
```

Comment:
whats with the extra brackets at top level?

Reply:
ah added them for all composite condition calls

Reply:
do these show up on the FE?

Reply:
ye :/ not sure where to remove extra brackets

Reply:
Hmm that's wierd. If we can't figure out why it's being sent, we should at least strip them in the FE.

Reply:
the only place I have brackets added is around any AND or OR s but im not checking if its just the composite conditions inside, could try do that or just get rid of the extra brackets at the end somewhere

Reply:
hmm actually @sonar3d i like this idea of checking if the brackets are needed or not by checking if the composite condition has more than one child.

Reply:
<img width="640" alt="image" src="https://user-images.githubusercontent.com/93496985/187135318-7394f736-ab7e-4e12-bca4-e139267e1475.png">
I added the bracket cleaner to the FE. It removes any extra parens that encompass the whole thing like `(smth smth smth )` but, leaves in those that are part of a composite condition like `(smth) AND (smth)`

Reply:
> 

will have a look at the children thing

Reply:
update had a look at children thing but is messyyy any suggestions welcome

Reply:
update #1000 this be scrapped cause code ew backend way - thanks Hussain for frontend help